portfolio/src/main/webapp/script.js

```diff
@@ -24,3 +24,19 @@ function closeNav() {
 }
 
 
+function getMessage() {
+  fetch('/data').then(response => response.text()).then((message) => {
+    document.getElementById('message-container').innerText = message;
+  });
+}
+
+
+function getMessageJSON() {
+  fetch('/data')  // sends a request to /data
+  .then(response => response.json()) // parses the response as JSON
+  .then((message) => { // now we can reference the fields in myObject!
+    console.log(message[0]);
```

Comment:
This is of course fine and you'll be turning this into DOM modification soon anyway, but a couple of things to note:

*   the console can log complex objects, so a single `console.log(message)` would work as well
*   it's a good practice to add some context to log messages, particularly as you get into more complex debugging scenarios. Something like `console.log("Messages received from server: " + message);` helps you know why you're seeing this information in a complex log
*   Another good practice is develop a convention for yourself for when you add "throwaway" log messages that you know you don't want to end up in production code. I prefix such log messages with my initials `MAC>> `. This makes it really easy to find them and remove them before submitting final code.

You're welcome to make as many changes as you want based on this comment (including zero :)). Mostly just some things to think about.